### ISSUE TITLE
When lora is loaded a second time,lora does not work

### ISSUE THREAD

**Issue by Daming-TF** [huggingface/diffusers#6524]
### Describe the bug

When loRA is loaded a second time,lora does not work

### Reproduction

```

import cv2
from PIL import Image
from insightface.app import FaceAnalysis
import torch
from diffusers import StableDiffusionPipeline, DDIMScheduler, AutoencoderKL
import os


def main():
    source_dir = r''
    base_model_path = fr'{source_dir}/Lykon--DreamShaper/'      # "SG161222/Realistic_Vision_V4.0_noVAE"
    vae_model_path = fr"{source_dir}/stabilityai--sd-vae-ft-mse"
    lora = f"{source_dir}/lora_1.5/Paper_Cutout.safetensors"
    device = "cuda"

    noise_scheduler = DDIMScheduler(
        num_train_timesteps=1000,
        beta_start=0.00085,
        beta_end=0.012,
        beta_schedule="scaled_linear",
        clip_sample=False,
        set_alpha_to_one=False,
        steps_offset=1,
    )
    vae = AutoencoderKL.from_pretrained(vae_model_path).to(dtype=torch.float16)
    pipe = StableDiffusionPipeline.from_pretrained(
        base_model_path,
        torch_dtype=torch.float16,
        scheduler=noise_scheduler,
        # unet=unet,
        vae=vae,
        feature_extractor=None,
        safety_checker=None
    ).to(device)
    pipe.load_lora_weights(
        os.path.dirname(lora), 
        weight_name=os.path.basename(lora), 
        adapter_name="style"
        )
    pipe.delete_adapters("style")
    pipe.load_lora_weights(
        os.path.dirname(lora), 
        weight_name=os.path.basename(lora), 
        adapter_name="style"
        )

    generator = torch.Generator(device).manual_seed(42)
    image = pipe(
            prompt='1 chicken',
            width=512, height=512,
            num_inference_steps=20,
            generator=generator,
        ).images[0]
    image.save('')


if __name__ == '__main__':
    main()
```

### System Info

diffusers-v0.25.0, Linux, python-3.10

### Who can help?

@sayakpaul @patrickvonplaten

**Comment by sayakpaul** [comment]
The code is not reproducible. Please help by providing a more minimal yet FULLY reproducible code snippet. Otherwise, we cannot look into it. 

Also, have tried calling `set_adapter()` after loading in the second LoRA?

Cc: @younesbelkada

**Comment by Daming-TF** [comment]
> import cv2
> from PIL import Image
> from insightface.app import FaceAnalysis
> import torch
> from diffusers import StableDiffusionPipeline, DDIMScheduler, AutoencoderKL
> import os
> 
> 
> def main():
>     source_dir = r''
>     base_model_path = fr'{source_dir}/Lykon--DreamShaper/'      # "SG161222/Realistic_Vision_V4.0_noVAE"
>     vae_model_path = fr"{source_dir}/stabilityai--sd-vae-ft-mse"
>     lora = f"{source_dir}/lora_1.5/Paper_Cutout.safetensors"
>     device = "cuda"
> 
>     noise_scheduler = DDIMScheduler(
>         num_train_timesteps=1000,
>         beta_start=0.00085,
>         beta_end=0.012,
>         beta_schedule="scaled_linear",
>         clip_sample=False,
>         set_alpha_to_one=False,
>         steps_offset=1,
>     )
>     vae = AutoencoderKL.from_pretrained(vae_model_path).to(dtype=torch.float16)
>     pipe = StableDiffusionPipeline.from_pretrained(
>         base_model_path,
>         torch_dtype=torch.float16,
>         scheduler=noise_scheduler,
>         # unet=unet,
>         vae=vae,
>         feature_extractor=None,
>         safety_checker=None
>     ).to(device)
>     pipe.load_lora_weights(
>         os.path.dirname(lora), 
>         weight_name=os.path.basename(lora), 
>         adapter_name="style"
>         )
>     pipe.delete_adapters("style")
>     pipe.load_lora_weights(
>         os.path.dirname(lora), 
>         weight_name=os.path.basename(lora), 
>         adapter_name="style"
>         )
> 
>     generator = torch.Generator(device).manual_seed(42)
>     image = pipe(
>             prompt='1 chicken',
>             width=512, height=512,
>             num_inference_steps=20,
>             generator=generator,
>         ).images[0]
>     image.save('')
> 
> 
> if __name__ == '__main__':
>     main()

```
from diffusers import DiffusionPipeline
import torch

pipeline = DiffusionPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16
).to("cuda")
pipeline.load_lora_weights("nerijs/pixel-art-xl", weight_name="pixel-art-xl.safetensors", adapter_name="pixel")
pipeline.delete_adapters("pixel")
pipeline.load_lora_weights("nerijs/pixel-art-xl", weight_name="pixel-art-xl.safetensors", adapter_name="pixel")
image = pipeline('1 chicken').images[0]
image.save('./debug.jpg')
```

**Comment by Daming-TF** [comment]
```
from diffusers import DiffusionPipeline
import torch

pipeline = DiffusionPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16
).to("cuda")
pipeline.load_lora_weights("nerijs/pixel-art-xl", weight_name="pixel-art-xl.safetensors", adapter_name="pixel")
pipeline.delete_adapters("pixel")
pipeline.load_lora_weights("nerijs/pixel-art-xl", weight_name="pixel-art-xl.safetensors", adapter_name="pixel")
image = pipeline('1 chicken').images[0]
image.save('./debug.jpg')
```

**Comment by Daming-TF** [comment]
Why does loRA not work when I reload the delete_adapter

**Comment by sayakpaul** [comment]
Indeed. Without calling `set_adapters()` we cannot see LoRA results. However, calling `set_adapters()` after deleting the adapters and loading LoRA weights seems to resolve the issue. 

Colab: https://colab.research.google.com/gist/sayakpaul/912bd85ccf009b72ba3db4a33018957b/scratchpad.ipynb

@younesbelkada @pacman100 is this behaviour expected?

**Comment by pacman100** [comment]
Hello,

When the only remaining lora is removed, the active adapter is set to an empty list. As such, when loading a new lora back, it needs to be set as active. This is a special case, otherwise, generally when you load a new lora, you would set the active adapter to it accordingly else the current active one will be used.  Would a warning like this "the only remaining lora was deleted and the active adapters is empty. When loading new adapter, set it to active adapter accordingly." help?
